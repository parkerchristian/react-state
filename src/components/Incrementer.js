import React, { PureComponent } from 'react';

export default class Incrementer extends PureComponent {
  state = {
    counter: 0
  };

  buttonClick = () => {
    this.setState(state => {
      return { counter: state.counter + 1 };
    });
  }

  render() {
    const { counter } = this.state;

    return (
      <section>
        <p>Count: {counter}</p>
        <button onClick={this.buttonClick}>Count!</button>
      </section>
    );
  }
}
