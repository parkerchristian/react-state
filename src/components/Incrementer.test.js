import React from 'react';
import { shallow } from 'enzyme';
import Incrementer from './Incrementer';

describe('Incrementer Tests', () => {
  it('Renders Incrementer', () => {
    const wrapper = shallow(<Incrementer />);
    expect(wrapper).toMatchSnapshot();
  });

  it('increments counter on click', () => {
    const wrapper = shallow(<Incrementer />);
    wrapper.find('button').at(0).simulate('click');
    expect(wrapper.state('counter')).toEqual(1);
  });

});
