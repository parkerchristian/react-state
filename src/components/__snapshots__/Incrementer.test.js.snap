// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Incrementer Tests Renders Incrementer 1`] = `
<section>
  <p>
    Count: 
    0
  </p>
  <button
    onClick={[Function]}
  >
    Count!
  </button>
</section>
`;
